### ISSUE TITLE
Switch camera while recording

### ISSUE THREAD

**Issue by blurtime** [eonist/HybridCamera#24]
Here a couple of resources regarding this issue. The first one sounds pretty nifty so that might be a good way to solve this problem: 

1. https://stackoverflow.com/questions/13951182/audio-video-out-of-sync-after-switch-camera/54770398#54770398
2. https://stackoverflow.com/questions/44577331/multiple-avcapturevideodataoutput-in-same-avcapturesession
3. https://stackoverflow.com/questions/47983572/switch-between-front-and-back-camera-while-recording-a-video

**Comment by blurtime** [comment]
Hi @eonist,
I found 4. today and thought it might be worth looking into. I played around with the code a bit but couldn't figure anything out - probably because the protocols for the buffering don't seem to be implemented in the project yet (unless I missed something) and one would have to rewrite a lot of code. Anyway, hope you're doing well and maybe have a little time to check it out. :D
Moritz

**Comment by eonist** [comment]
Hey man, packed at work at the moment. Switching channels while recording is a deep rask. Aka requires deep focus. Its exciting to see Apple embracing this feature in their new camera app. If you havnt seen the Apple event, it had a segment about this feature. Even recording both cams at the same time.

**Comment by blurtime** [comment]
I'll watch it right away! Thank you! No worries, it's not that big of a deal :)

**Comment by eonist** [comment]
If I can't add features my self, I'll probably start funding them through issuehunt.

**Comment by eonist** [comment]
Yeah, I think I focused a lot on nailing the zero-lag part. I just did deep focus for a week to get the core stuff right. Then refactored now and then, and people added code etc. Like you. I will start funding this project next month or start improving it my self. Either or. Stay tuned ;)

**Comment by blurtime** [comment]
You really did nail that part! Couldn't imagine using any other library exactly for this fact! 
Really looking forward to improvements (especially or currently only to switching while recording)! Maybe one could come up with some special ideas like filters (or something crazy), too, which would make this library stick out to an even greater extent - if there's time obviously. 
I'll probably launch my app in April and will be very happy to use this library 😌😁

**Comment by eonist** [comment]
Filters is super easy. Lets do that next. I did a code quality refactor yesterday, thinking to Add. Etter serie handling with Result today. And then Add AutoLayout based ui in i fresh scheme.

**Comment by blurtime** [comment]
Sounds awesome, looking forward to it! 😃 My next semester will start next week so I'm not sure how much time I will have. If I do have some time on my hands, I'll hit you up and try to help out :)

**Comment by eonist** [comment]
Added support for switching the different backchat types. Its just a start. Working on support for triple camera module that iPhone 11 pro has now.
